analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/RequestsWithExcessiveLengthBase.cs

```diff
@@ -59,28 +62,92 @@ protected RequestsWithExcessiveLengthBase(System.Resources.ResourceManager rspec
+                        Language.SyntaxKind.Attribute);
+
+                    c.RegisterCompilationEndAction(cc => ReportOnCollectedAttributes(cc, attributesOverTheLimit));
+                });
+
+        protected bool IsRequestFormLimits(string attributeName) =>
+            attributeName.Equals(RequestFormLimits, Language.NameComparison) || attributeName.Equals(RequestFormLimitsAttribute, Language.NameComparison);
+
+        protected bool IsRequestSizeLimit(string attributeName) =>
+            attributeName.Equals(RequestSizeLimit, Language.NameComparison) || attributeName.Equals(RequestSizeLimitAttribute, Language.NameComparison);
+
+        private void CollectAttributesOverTheLimit(SyntaxNodeAnalysisContext context, ConcurrentDictionary<SyntaxNode, Attributes> attributesOverTheLimit)
+        {
+            if (!IsEnabled(context.Options))
+            {
+                return;
+            }
+
+            var attribute = (TAttributeSyntax)context.Node;
+
+            if (attribute.IsKnownType(KnownType.Microsoft_AspNetCore_Mvc_DisableRequestSizeLimitAttribute, context.SemanticModel))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, attribute.GetLocation()));
+                return;
+            }
+
+            var requestSizeLimit = IsInvalidRequestSizeLimit(attribute, context.SemanticModel);
+            var requestFormLimits = IsInvalidRequestFormLimits(attribute, context.SemanticModel);
+
+            if ((requestSizeLimit != null
+                 || requestFormLimits != null)
+                && GetMethodOrClassDeclaration(attribute, context.SemanticModel) is { } declaration)
+            {
+                attributesOverTheLimit.AddOrUpdate(
+                    declaration,
+                    new Attributes(requestFormLimits, requestSizeLimit),
+                    (declaration, attributes) => new Attributes(requestFormLimits, requestSizeLimit, attributes));
+            }
+        }
+
+        private void ReportOnCollectedAttributes(CompilationAnalysisContext context, ConcurrentDictionary<SyntaxNode, Attributes> attributesOverTheLimit)
+        {
+            foreach (var attrsByDec in attributesOverTheLimit)
+            {
+                context.ReportDiagnosticWhenActive(
+                    attrsByDec.Value.SecondaryAttribute != null
+                        ? Diagnostic.Create(rule, attrsByDec.Value.MainAttribute.GetLocation(), new List<Location> { attrsByDec.Value.SecondaryAttribute.GetLocation() })
+                        : Diagnostic.Create(rule, attrsByDec.Value.MainAttribute.GetLocation()));
+            }
+        }
 
         private bool IsEnabled(AnalyzerOptions options)
         {
             analyzerConfiguration.Initialize(options);
             return SupportedDiagnostics.Any(d => analyzerConfiguration.IsEnabled(d.Id));
         }
+
+        private struct Attributes
+        {
+            private readonly SyntaxNode invalidRequestForm;
```

Comment:
I suggest to name them "requestForm" and "requestSize"

this struct doesn't ensure they are invalid or not - it can hold valid attributes as well